nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -0,0 +1,535 @@
+{
+  try {
+    auto node = shared_from_this();
+
+    // Initializing polygon topic with empty string.
+    // If it won't be re-writed, polygons won't be published.
+    nav2_util::declare_parameter_if_not_declared(
+      node, "polygon_topic", rclcpp::ParameterValue(""));
+    polygon_topic_ = get_parameter("polygon_topic").as_string();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "odom_topic", rclcpp::ParameterValue("odom"));
+    odom_topic_ = get_parameter("odom_topic").as_string();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "cmd_vel_in_topic", rclcpp::ParameterValue("cmd_vel_raw"));
+    cmd_vel_in_topic_ = get_parameter("cmd_vel_in_topic").as_string();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "cmd_vel_out_topic", rclcpp::ParameterValue("cmd_vel"));
+    cmd_vel_out_topic_ = get_parameter("cmd_vel_out_topic").as_string();
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, "base_frame_id", rclcpp::ParameterValue("base_footprint"));
+    base_frame_id_ = get_parameter("base_frame_id").as_string();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "transform_tolerance", rclcpp::ParameterValue(0.1));
+    double transform_tolerance = get_parameter("transform_tolerance").as_double();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "max_time_shift", rclcpp::ParameterValue(2.0));
+    max_time_shift_ = get_parameter("max_time_shift").as_double();
+    nav2_util::declare_parameter_if_not_declared(
+      node, "simulation_time_step", rclcpp::ParameterValue(0.02));
+    double simulation_time_step = get_parameter("simulation_time_step").as_double();
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, "release_step", rclcpp::ParameterValue(0.002));
+    release_step_ = get_parameter("release_step").as_double();
+
+    // Leave these parameters to be not initialized:
+    // the will cause an error if it will not set
+    nav2_util::declare_parameter_if_not_declared(
+      node, "polygons", rclcpp::PARAMETER_STRING_ARRAY);
+    nav2_util::declare_parameter_if_not_declared(
+      node, "observation_sources", rclcpp::PARAMETER_STRING_ARRAY);
+
+    // Working with polygons parameters
+    polygon_names_ = get_parameter("polygons").as_string_array();
+    for (std::string polygon_name : polygon_names_) {
+      // Leave it not initialized: the will cause an error if it will not set
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name + ".type", rclcpp::PARAMETER_STRING);
+      const std::string polygon_type = get_parameter(polygon_name + ".type").as_string();
+
+      if (polygon_type == "polygon") {
+        std::shared_ptr<Polygon> p = std::make_shared<Polygon>(
+          node, polygon_name, simulation_time_step);
+
+        if (!p->getParameters()) {
+          return false;
+        }
+
+        polygons_.push_back(p);
+      } else {  // polygon_type == "circle"
```

Comment:
else if, check if circle, in case someone typed in something nonsensible and throw an exception if it's not either circle or polygon

Reply:
Fixed for polygons and sources.